File: Java Leetcode/Solution41.java
// 41. First Missing Positive

import java.util.Arrays;

public class Solution41 {

    public int firstMissingPositive(int[] nums) {
        int[] filteredNums = Arrays.stream(nums).filter(n -> n > 0).toArray();

        Arrays.sort(filteredNums);

        int target = 1;
        for (int n : filteredNums) {
            if (n == target) {
                target++;
            } else if (n > target) {
                return target;
            }
        }
        return target;
    }
}

// class Solution {
//     public int firstMissingPositive(int[] nums) {
//         int n = nums.length;
//         for(int i =0;i<n;i++){
//             int num = nums[i];
//             while(nums[i]<=n && nums[i]>0 && nums[num-1]!=num){
//                 int temp = nums[i];
//                 nums[i]=nums[num-1];
//                 nums[num-1]=temp;
//                 num = nums[i];
//             }
//         }
//         for(int i=0;i<n;i++){
//             if(nums[i]!=i+1){
//                 return i+1;
//             }
//         }
//         return n+1;
//     }
// }

File: Python Leetcode/Solution41.py
# 41. First Missing Positive

class Solution41:
    def firstMissingPositive(self, nums):
        nums = [n for n in nums if n > 0]
        nums.sort()

        target = 1
        for n in nums:
            if n == target:
                target += 1
            elif n > target:
                return target
            
        return 
    

"""
class Solution:
    def firstMissingPositive(self, nums: List[int]) -> int:
        n=len(nums)
        seen=[False]*(n+1)
        for num in nums:
            if 0<num<=n:
                seen[num]=True
        for i in range(1,n+1):
            if seen[i]==False:
                return i
        return n+1
"""

